lightning/src/ln/channelmanager.rs

```diff
@@ -6324,23 +6324,24 @@ where
 					}
 				}
 
-				let htlcs = payment.htlcs.iter().map(events::ClaimedHTLC::from).collect();
-				let sender_intended_value = payment.htlcs.first().map(|htlc| htlc.total_msat);
-				let dup_purpose = claimable_payments.pending_claiming_payments.insert(
-					payment_hash,
-					ClaimingPayment {
-						amount_msat: payment.htlcs.iter().map(|source| source.value).sum(),
-						payment_purpose: payment.purpose,
-						receiver_node_id,
-						htlcs,
-						sender_intended_value,
-					},
-				);
-				if dup_purpose.is_some() {
-					debug_assert!(false, "Shouldn't get a duplicate pending claim event ever");
-					log_error!(self.logger, "Got a duplicate pending claimable event on payment hash {}! Please report this bug",
-						&payment_hash);
-				}
+				let _claiming_payment = claimable_payments.pending_claiming_payments
+					.entry(payment_hash)
+					.and_modify(|_| {
+						debug_assert!(false, "Shouldn't get a duplicate pending claim event ever");
+						log_error!(self.logger, "Got a duplicate pending claimable event on payment hash {}! Please report this bug",
+							&payment_hash);
+					})
```

Comment:
Noted that there is a small change of behavior here. When this case is hit, the entry is not overridden.

Reply:
IIUC, this is actually an invariant check and should never happen. We'll only insert into `ClaimablePayments::claimable_payments` if `ClaimablePayments::pending_claiming_payments` doesn't contain the same payment hash.

https://github.com/lightningdevkit/rust-lightning/blob/df0120809f3453aa4cfc6ca4d34bbedcc823928d/lightning/src/ln/channelmanager.rs#L5455-L5466

So I _think_ we don't allow two outstanding payments for the same payment hash, but we will allow consecutive ones.

cc: @TheBlueMatt for confirmation.

Maybe `ClaimablePayments` should have been one map storing an enum?